openstack/baremetal/v1/nodes/configdrive.go

```diff
@@ -0,0 +1,120 @@
+package nodes
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+// A ConfigDrive struct will be used to create a base64-encoded, gzipped ISO9660 image for use with Ironic.
+type ConfigDrive struct {
```

Comment:
Since a "util" function has a lot more leniency than a core Gophercloud function, feel free to add additional fields here that might be useful. For example, `TemporaryDirectory` or `BuildDirectory` where `TempDir` and `TempFile` can use as the base directory.